src/xSTIR/pSTIR/STIR.py

```diff
@@ -482,7 +486,7 @@ def set_modality(self, mod):
     def __array_interface__(self):
         """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
         if not self.contiguous:
-            raise ContiguousError("please make a copy first with `as_array`")
+            raise ContiguousError("please make an array-copy first with `as_array`")
```

Comment:
not sure I understand this message: you end up here if you do `image.asarray` for `image` with non-contiguous data, in which case we should recommend the user to work with `image_copy = image.copy()` instead

Reply:
> you end up here if you do `image.asarray` for `image` with non-contiguous data

agreed, i.e. the user wants a `numpy.ndarray`.

> in which case we should recommend the user to work with `image_copy = image.copy()` instead

but that would not give the user a `numpy.ndarray`.

Reply:
not sure I follow. Here's my intention:

`method() -> dtype` |contiguous|discontiguous
-|-|-
`__array_interface__ -> dict` | view | `ValueError`
`asarray() -> xp.ndarray` | view | copy
`as_array() -> xp.ndarray` | copy (and in future, warn) | copy

Reply:
My preference is with constructing contiguous data objects. This can be done right now by Python constructors (I actually did it, and it worked) and eventually in C++.

The advantage of this approach is that any Python script that works today will work verbatim in the future.

Reply:
Do you mean you've already fixed https://github.com/UCL/STIR/issues/1468 ?

Reply:
> Do you mean you've already fixed [UCL/STIR#1468](https://github.com/UCL/STIR/issues/1468) ?

yes https://github.com/UCL/STIR/pull/1592, but `resize/grow` still to do.

Reply:
> ... just a temporary quick fix - eventually we must make all our `ImageData` constructors deliver objects storing contiguous data, which would make all contiguity checks and recommendations we are discussing now obsolete.

not really. What about `NiftiImagedata`? (should be doable), `GadgetronImageData` (a pain?), `GadgetronAcquisitionData` (impossible?).

> The advantage of this approach is that any Python script that works today will work verbatim in the future.

not really. Users mgith expect that it's a copy, not a view. i.e. modified the ndarray, but not expect mods to the SIRF object.

Reply:
> `GadgetronAcquisitionData` (impossible?).

Ah so we're missing a `@propery contiguous` on `AcquisitionData` too?

Reply:
neither `NityImageData` nor `Gadgetron` data containers have `asarray` method

Reply:
should be in `DataContainer` I guess

Reply:
the default implementation, `supports_array_view` should be false by default. Then everything should work ok. (i.e. raise an error, unless asking for a copy)

Reply:
Or if you mean you want some array functionality added to `NityImageData` & `Gadgetron` data containers, that's for a different PR methinks.